accessible/tree.service.js

```diff
@@ -0,0 +1,305 @@
+              nextNode = nextNode.parentNode;
+            }
+            while (nextNode && nextNode.className != 'treeview' && nextNode.tagName != 'LI') {
+              nextNode = nextNode.parentNode;
+            }
+            if (nextNode.className == 'treeview' || nextNode == null){
+              return;
+            }
+            this.updateSelectedNode(nextNode, tree);
+            break;
+          case 38:
+            //up-facing arrow: go up a level, if possible. If not, do nothing
+            e.preventDefault();
+            e.stopPropagation();
+            console.log('node passed in: ' + node.id);
+            var prevSibling = this.getPreviousSibling(node);
+            if (prevSibling && prevSibling.tagName != 'H1'){
+              this.updateSelectedNode(prevSibling, tree);
+            } else {
+              console.log('no previous sibling');
+            }
+            break;
+          case 39:
+            e.preventDefault();
+            e.stopPropagation();
+            console.log('in right arrow section');
+            var firstChild = this.getFirstChild(node);
+            if (firstChild){
+              this.updateSelectedNode(firstChild, tree);
+            } else {
+              console.log('no valid child');
+            }
+            break;
+          case 40:
+            //down-facing arrow: go down a level, if possible. If not, do nothing
+            console.log('preventing propogation');
+            e.preventDefault();
+            e.stopPropagation();
+            var nextSibling = this.getNextSibling(node);
+            if (nextSibling){
+              this.updateSelectedNode(nextSibling, tree);
+            } else {
+              console.log('no next sibling');
+            }
+            break;
+          case 13:
+            //if I've pressed enter, I want to interact with a child
+            console.log('enter is pressed');
+            var activeDesc = this.getActiveDesc(treeId);
+            // if (activeDesc.tagName == 'BUTTON') {
+            //   activeDesc.click();
+            // }
+            // if (activeDesc.tagName == 'INPUT'){
+            //       activeDesc.focus();
+            // }
+            if (activeDesc){
+              var children = activeDesc.children;
+              var child = children[0];
+              if (children.length == 1 && (child.tagName == 'INPUT' || child.tagName == 'BUTTON')){
+                if (child.tagName == 'BUTTON'){
+                  child.click();
+                }
+                if (child.tagName == 'INPUT'){
+                  child.focus();
+                }
+              }
+            } else {
+              console.log('no activeDesc');
+            }
+        }
+        this.previousKey_=e.keyCode;
+      },
+      getFirstChild:  function(element){
+        if (element == null){
+          return element;
+        } else {
+          var childList = element.children;
+          for (var i=0; i<childList.length; i++){
```

Comment:
Use Iterating over Node Lists trick here for more efficient code.
https://google.github.io/styleguide/javascriptguide.xml?showone=Tips_and_Tricks#Tips_and_Tricks